src/dbnode/storage/index.go

```diff
@@ -987,13 +1012,25 @@ func (i *nsIndex) query(
 	opts index.QueryOptions,
 	execBlockFn execBlockQueryFn,
 ) (bool, error) {
+	ctx, sp := ctx.StartTraceSpan(tracepoint.NSIdxSubQuery)
+	sp.LogFields(
+		opentracinglog.String("query", query.String()),
+		opentracinglog.String("namespace", i.nsMetadata.ID().String()),
+		opentracinglog.Int("limit", opts.Limit),
+		xopentracing.Time("start", opts.StartInclusive),
+		xopentracing.Time("end", opts.EndExclusive),
+	)
+	defer sp.Finish()
+
 	// Capture start before needing to acquire lock.
 	start := i.nowFn()
 
 	i.state.RLock()
 	if !i.isOpenWithRLock() {
 		i.state.RUnlock()
-		return false, errDbIndexUnableToQueryClosed
+		err := errDbIndexUnableToQueryClosed
+		sp.LogFields(opentracinglog.Error(err))
```

Comment:
Hey can we perhaps just make a method to handle this for us?  Perhaps put it in the `context` package?

```go
func LogSpanErrorAndReturn(sp opentracing.Span, err error) error {
  sp.LogFields(opentracinglog.Error(err))
  return err
}
```

Then you can just:
```go
  return false, context.LogSpanErrorAndReturn(err)
```